CIP-0034/README.md

```diff
@@ -0,0 +1,82 @@
+---
+CIP: 34
+Title: Smart Contract Software Licensing
+Authors: Pi Lanningham <pi@sundaeswap.finance>
+Comments-URI:
+Status: Draft
+Type: Informational
+Created: 2022-01-03
+Post-History: 
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This proposal defines a metadata standard for specifying the license for a smart contract.
+
+## Motivation
+
+Smart contracts, specified in source code or compiled to raw bytecode as presented on the blockchain, are creative and often highly valuable works and are often subject to legal licensing concerns.  It is important that companies building on Cardano have the ability to link their smart contracts unambiguously to the license.
```

Comment:
> are often subject to legal licensing concerns

Is this in fact the case? You put your smart contract on the blockchain, and now everyone has it. What happens if I take a random script from my local copy, and put it on my website? You could sue me, but I think in many jurisdictions, you'd have a hard time winning the suit, especially when this is what blockchain explorers etc. are already doing.

Reply:
Of course, this doesn't apply to the *source form*, but that's not relevant to this CIP, because that's not put on the chain, and if it is, the same concerns as above apply.

Reply:
I would assume that attaching the license only matters when you are providing/using the source code; for example, when you are submitting transactions and providing the source or using reference inputs. Would it make more sense then to attach the metadata to the Plutus script instead? There could be a standard for wrapping a Plutus script, which would encode a license without adding greatly to execution costs.